src/UseCases/AddDonation/AddDonationUseCase.php

```diff
@@ -58,9 +59,14 @@ public function addDonation( AddDonationRequest $donationRequest ) {
 			return ValidationResponse::newFailureResponse( $validationResult->getViolations() );
 		}
 
-		$this->donationRepository->storeDonation( $donation );
+		$needsModeration = $this->donationValidator->needsModeration( $donation );
+		if ( $needsModeration ) {
+			$donation->setStatus( Donation::STATUS_MODERATION );
+		}
+
+		$donationId = $this->donationRepository->storeDonation( $donation );
 
-		$this->sendDonationConfirmationEmail( $donation );
+		$this->sendDonationConfirmationEmail( $donationId, $donation, $needsModeration );
```

Comment:
Ah... this is silly. The Donation domain object should have the id. Which makes it clear the comment that was in the repository was not quite accurate, as it needs to update the passed donation object, not return the id. (That can be changed in another commit though (kinda unrelated to what you are doing here), and I'm happy to do this.)